src/udev/udev-watch.c

```diff
@@ -360,6 +360,7 @@ int manager_init_inotify(Manager *manager, int fd) {
         manager->inotify_fd = fd;
         (void) udev_watch_restore(manager);
 
+        (void) notify_remove_fd_warn("inotify");
         r = notify_push_fd(manager->inotify_fd, "inotify");
```

Comment:
hmm, this smells like an issue we always should think about when using notify_push_fd(). Maybe change notify_push_fd() to take a bool parameter "overwrite" or so, which when true sends a remove message first. With that we are forced about to think about the problem each time we use the helper.

Reply:
Let's do that unconditionally, as current usecases assume that the name of each fd is unique.